fili-core/src/main/java/com/yahoo/bard/webservice/data/config/table/MetricUnionCompositeTableDefinition.java

```diff
@@ -93,7 +99,12 @@ public PhysicalTable build(ResourceDictionaries dictionaries, DataSourceMetadata
     private Set<PhysicalTable> getPhysicalTables(ResourceDictionaries resourceDictionaries) {
         PhysicalTableDictionary physicalTableDictionary = resourceDictionaries.getPhysicalDictionary();
         return dependentTableNames.stream()
-                .map(name -> physicalTableDictionary.getOrDefault(name, null))
+                .peek(name -> {
```

Comment:
Instead of doing this to every element once, maybe just check if the resulting table collection is the same size as the dependent table names? I guess its a preference thing, but it feels like size checking is a bit simpler. Also the error message should be a bit more verbose saying that one of the dependent table is missing when building this metric union composite table.